packaging/genere-views.cmake

```diff
@@ -0,0 +1,78 @@
+# Launch genere binary
+# cmake-format: off
+# The funtion api is almost the same as genere:
+#
+# Genere <ClassName> <ClassLabel> <AttributeFileName> <LogFile> [options]
+#   <ClassName>: base name for generated classes
+#   <ClassLabel>: label for generated classes
+#   <AttributeFileName>: name of the file (.dd) containing the attribute specifications
+#   <LogFile>: name of the log file, it must be included in the source files of the binary/library
+#
+# Options:
+#   -nomodel no generation of class <ClassName>
+#   -noarrayview no generation of class <ClassName>ArrayView
+#   -noview no generation of classes <ClassName>View and <ClassName>ArrayView
+#   -nousersection no generation of user sections
+#   -specificmodel <SpecificModelClassName> name of a specific model class, to use instead of ClassName
+#   -super <SuperClassName> name of the parent class
+#
+# Do nothing if GENERATE_VIEWS is set to OFF
+#
+# Example to use genere binary in CMakeLists.txt:
+#
+#   include(genere)
+#   generate_gui(FOO "Foo Example" Foo.dd Foo.dd.log -noarrayview)
+#   add_executable(basetest Foo.cpp) 
+#   if(GENERATE_VIEWS)
+#       target_sources(basetest PRIVATE Foo.dd.log) # Foo.dd.log is genereted by generate_gui
+#   endif()
+#   generate_gui_add_dependency(basetest)
+#
+# The log file Foo.dd.log is generated by generate_gui and must be added as a source of the target. 
+# Whithout it, generate_gui is never triggered.
+#
```

Comment:
It should be explained somewhere that the log files act like dummy targets. And that this is necessary to autoupdate the sources without erasing them.

Reply:
I think it should be explained why we have to add the log files as sources since it is rather tricky:
- We cannot regenerate from zero the h/cpp view files because they are inputs and outputs
- So we use the logs as dummy targets for the main target.
This way we have the desired mechanic: The views are generated only when `GENERATE_VIEWS=ON` and a `.dd` file was modified.

Reply:
done